Mathlib/Geometry/Manifold/Bordisms.lean

```diff
@@ -0,0 +1,211 @@
+/-
+Copyright (c) 2024 Michael Rothgang. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Michael Rothgang
+-/
+import Mathlib.Geometry.Manifold.InteriorBoundary
+import Mathlib.Geometry.Manifold.ContMDiff.Defs
+
+/-!
+## (Unoriented) bordism theory
+
+This file defines the beginnings of (unoriented) bordism theory. For the full definition,
+a number of prerequisites are missing from mathlib, but a surprising amount of progress
+can already be made today.
+
+Currently, this file only contains the definition of *singular *n*-manifolds*:
+bordism classes are the equivalence classes of singular n-manifolds w.r.t. the (co)bordism relation.
+
+## Main definitions
+
+- **SingularNManifold**: a singular `n`-manifold on a topological space `X`, for `n ∈ ℕ`, is a pair
+`(M,f)` of a closed `n`-dimensional smooth manifold `M` together with a continuous map `M → X`.
+- `SingularNManifold.map`: a map `X → Y` of topological spaces induces a map between the spaces
+of singular n-manifolds
+- `SingularNManifold.comap`: if `(N,f)` is a singular n-manifold on `X` and `φ: M → N` is smooth,
+the `comap` of `(N,f)` and `φ` is the induced singular n-manifold `(M, f ∘ φ)` on `X`.
+- `SingularNManifold.empty`: the empty set `M`, viewed as an `n`-manifold,
+as a singular `n`-manifold over any space `X`
+- `SingularNManifold.trivial`: an `n`-dimensional manifold induces a singular `n`-manifold
+on the one-point space.
+- `SingularNManifold.prod`: the product of a singular `n`-manifold and a singular `m`-manifold
+on the one-point space, is a singular `n+m`-manifold on the one-point space.
+
+## TODO
+- define cobordisms and the cobordism relation
+- prove that the cobordisms relation is an equivalence relation
+- define unoriented bordisms groups (as a set of equivalence classes),
+prove they are a group
+- prove that unoriented bordism groups define an extraordinary homology theory
+
+## Tags
+singular n-manifold, cobordism
+
+-/
+
+open scoped Manifold
+open FiniteDimensional Set
+
+noncomputable section
+
+universe u in
+/-- A **singular `n`-manifold** on a topological space `X`, for `n ∈ ℕ`, is a pair `(M,f)`
+of a closed `n`-dimensional smooth manifold `M` together with a continuous map `M → X`. -/
+structure SingularNManifold (X : Type*) [TopologicalSpace X] (n : ℕ) where
+  /-- The normed space on which the manifold `M` is modeled. -/
+  E : Type u
```

Comment:
I don't think it can be correct to assume that `E` lives in the same universe as `M`: typically, I expect `M` to live in a generic universe, while `E` would be `R^n` (living in `Type`). So your definition should probably depend on two universes `u` (for `M`) and `v` (for `E` and `H`)?